job-server-api/src/main/scala/spark/jobserver/japi/JavaJob.scala

```diff
@@ -0,0 +1,30 @@
+package spark.jobserver.japi
+
+import com.typesafe.config.Config
+import org.scalactic._
+import spark.jobserver.api._
+
+import scala.reflect.ClassTag
+
+case class JavaValidationException(reason: String) extends Exception with ValidationProblem
+
+case class JavaJob[R: ClassTag, CX: ClassTag](job: BaseJavaJob[R, CX]) extends SparkJobBase {
+
+  import spark.jobserver.japi.{JavaValidationException => JVE}
+
+  override type C = CX
+  override type JobData = Config
+  override type JobOutput = R
+
+  def runJob(sc: CX, runtime: JobEnvironment, data: Config): R = {
+    job.run(sc, runtime, data)
+  }
+
+  def validate(sc: CX, runtime: JobEnvironment, config: Config): Config Or Every[ValidationProblem] = {
+    job.verify(sc, runtime, config) match {
+      case j : JVE => Bad(One(j))
+      case t : Throwable => Bad(One(JVE(t.getMessage)))
```

Comment:
We will be loosing some important stack trace information here, consider adding t `JVE(t.getMessage, t)`